### PR TITLE
PR: Close terminal tab if process was closed

### LINKED ISSUE #95: Close automatically a Terminal tab if it was closed from the process itself
Right now, if ``exit`` or Ctrl+D is executed to finish a terminal process, we output the message ``Pipe closed`` and the tab remains open. We would like to override this behaviour by closing the tab when the process is finished.

### LINKED ISSUE #96: TypeError: 'Icon' object is not callable
_From @wlsdnen on July 26, 2017 12:59_

## Description

**What steps will reproduce the problem?**

1. 
2. 
3. 

**What is the expected output? What do you see instead?**

**Please provide any additional information below**

  File "c:\users\jjw\appdata\local\programs\python\python35\lib\site-packages\spyder_terminal\terminalplugin.py", line 156, in __wait_server_to_start
    QMessageBox.Ok)
TypeError: 'Icon' object is not callable

## Version and main components

* Spyder Version: 3.2.0 
* Python Version: 3.5.3
* Qt Versions: 5.9.1, PyQt5 5.9 on Windows

## Dependencies
```
pyflakes >=0.6.0 :  1.5.0 (OK)
pycodestyle >=2.3:  2.3.1 (OK)
pygments >=2.0   :  2.2.0 (OK)
pandas >=0.13.1  :  None (NOK)
numpy >=1.7      :  1.13.1 (OK)
sphinx >=0.6.6   :  1.6.3 (OK)
rope >=0.9.4     :  0.10.5 (OK)
jedi >=0.9.0     :  0.10.2 (OK)
nbconvert >=4.0  :  5.2.1 (OK)
sympy >=0.7.3    :  None (NOK)
cython >=0.21    :  0.26 (OK)
qtconsole >=4.2.0:  4.3.0 (OK)
IPython >=4.0    :  6.1.0 (OK)
pylint >=0.25    :  1.7.2 (OK)

```

_Copied from original issue: spyder-ide/spyder#4817_

### PR DESCRIPTION
**Pull request by andfoy**
Fixes #96 
Fixes #95

### CODE REVIEW

File: spyder_terminal/tests/test_terminal.py [hunk 1]
```diff
@@ -131,3 +134,25 @@ def test_output_redirection(qtbot):
     stderr = osp.join(getcwd(), 'spyder_terminal_err.log')
     assert osp.exists(stdout) and osp.exists(stderr)
     terminal.closing_plugin()
+
+
+def test_close_terminal_manually(qtbot):
+    # Setup widget
+    terminal = setup_terminal(qtbot)
+    qtbot.wait(TERM_UP)
```
Comment (goanpeca) [inline spyder_terminal/tests/test_terminal.py]: @andfoy is not there a signal that gets emitted when the terminal is ready?

Comment (goanpeca) [inline spyder_terminal/tests/test_terminal.py]: I thought that we had some mechanism in place to test when a terminal was up, and then emitted a signal when confirmed.

Comment (andfoy) [inline spyder_terminal/tests/test_terminal.py]: @goanpeca I tried to put some kind of mechanism on place, however I didn't put it on place. Right now I have a way to know if the prompt is ready, I'll try to use it to emit a signal.

Comment (goanpeca) [inline spyder_terminal/tests/test_terminal.py]: @andfoy 👍 1:

Comment (andfoy) [inline spyder_terminal/tests/test_terminal.py]: @goanpeca If I use the ready signal, some tests start to fail randomly. I prefer to leave them as they are

Comment (goanpeca) [inline spyder_terminal/tests/test_terminal.py]: Ok, then, but in this case I think is better that the test has a timer that check every 1000 ms, and we define a maximum number of retries, to account for changes in CI times etc...

**Pull request merged**